lib/generate_letters.ml

```diff
@@ -9,34 +9,31 @@ type letter_deck = char list
     char*)
 let random_let () = Char.chr (Random.int 26 + 65)
 
-(** [start_game lst] is a letter_deck of 7 chars. Requires lst to be
-    empty since it is the start of the game. start_game : unit ->
-    letter_deck *)
-let rec start_game = function
-  | [] -> start_game [ random_let () ]
-  | h :: t ->
-      if List.length t = 6 then h :: t
-      else start_game (h :: random_let () :: t)
+(** [new_deck ()] returns a letter_deck of 7 chars. *)
+let new_deck () =
+  let rec new_deck_inner = function
+    | [] -> new_deck_inner [ random_let () ]
+    | h :: t ->
+        if List.length t = 6 then h :: t
+        else new_deck_inner (h :: random_let () :: t)
+  in
+  new_deck_inner []
+
+exception Missing_letter
 
 (** [remove_let curr goal lst] removes the appropriate letter when
     players use a letter and refills it with a new random letter.
     Requires curr to be the current index as the function progresses
     (starting at 0) and goal to be the index of the letter to be
     removed. remove_let : int -> int -> letter_deck -> letter_deck *)
-let rec remove_let curr goal = function
-  | [] -> start_game []
-  | h :: t ->
-      if curr = goal then random_let () :: t
-      else h :: remove_let (curr + 1) goal t
-
-(** [index_of_letter curr letter lst] is the index of the first
-    occurrence of this character in the list. Requires letter to be
-    present in lst and curr to be the current index (starting at 0).
-    index_of_letter : int -> char -> letter_deck -> int*)
-let rec index_of_letter curr letter = function
-  | [] -> raise (Failure "Empty List")
+let rec replace_letter letter letter_deck =
```

Comment:
Same thing. There is no longer a [remove_let] function, the .mli documents it as [replace_let_biased].